File: app/assets/javascripts/flash_helper.js
var close_btn = '<button class="close" data-dismiss="alert">&times;</button>';
var speed = 4500;

$(document).ready(function() {
  
  // Remove the flash notice on click
  $('#flash_helper > div').live('click', function(e) {
    $(this).fadeOut(function() {
      $(this).hide();
    });
  });

  // Automatically remove the div after 4.5 seconds
  $('#flash_helper > div').delay(speed).fadeOut('fast', function(){
    $(this).remove();
  });

});

// Utility methods for the flash_helper
var flash_helper = {
  
  // Dynamically create a flash notice
  create: function(message, type){
    var type = type ? "alert-" + type : "alert-success";
    var $new_message = $('<div class="alert ' + type + '">' + message + close_btn + '</div>');
    $new_message.delay(speed).fadeOut('fast', function(){
      $(this).remove();
    });
    $('#flash_helper').append($new_message)
  },

  // Clear all the flash notices
  clear: function() {
    $('#flash_helper').html('');
  },

};

File: app/helpers/flash_helper.rb
module FlashHelper

  # Flash Helper
  # ------------------------------------------------------------------------------------------------------

  # a robust method to display flash messages
  def flash_helper
    # html from each message
    html = ''
    flash.each do |key, msg|
      html = html + content_tag(:div, raw(msg + close_alert_btn), :class => "alert alert-#{key}")
    end

    return (content_tag :div, html.html_safe, id: "flash_helper").html_safe
  end

  # Generate the html for a close button on alerts and other modules
  def close_alert_btn type="alert"
    content_tag :button, "&times;".html_safe, :class => "close", :'data-dismiss' => type
  end

end